naga/src/valid/handles.rs

```diff
@@ -180,6 +183,19 @@ impl super::Validator {
             validate_type(ty)?;
         }
 
+        {
+            for (handle, _node) in diagnostic_filters.iter() {
+                handle.check_valid_for(diagnostic_filters)?;
+                let DiagnosticFilterNode { inner: _, parent } = diagnostic_filters[handle];
+                if let Some(handle) = parent {
+                    handle.check_valid_for(diagnostic_filters)?;
```

Comment:
Ah, yes, that would be better, wouldn't it? Each node's parent already gets checked in the iteration pass, so I should only need to call `diagnostic_filter_leaf.check_valid_for(…)` at the end.

Oh...come to think of it, I shouldn't need to do `handle.check_valid_for(…)` here, because the fact that the handle comes from an iterator over the arena makes the check redundant. 😅